### ISSUE TITLE
 urgent question,please

### ISSUE THREAD

**Issue by zuoeye** [jwyang/JULE.torch#25]
There is only class in my clustering result   when I run my data（4 classes).

 It is an urgent question. Please help me as soon as possible. 

Thank you! I look forward to your reply as soon as possible！

Thank you！

**Comment by jwyang** [comment]
Hi, @zuoeye , did you specify the number of clusters at the beginning?

**Comment by zuoeye** [comment]
How to specify the numer  of clusters at the beginning???

Thank you!

**Comment by jwyang** [comment]
please see line 163 in train.lua

**Comment by zuoeye** [comment]
target_nclusters_table[i] = #(label_gt_table_table[i]) ?

 Is this function‘s goal to get the clusters number of clustering?? This is not to specify the numer of

 clusters at the beginning.

Thank you

**Comment by zuoeye** [comment]
Does I specify the number of target clusters ？ For example，should I specify four target clusters at the begining if I want to get four categories？

Thank you！

**Comment by jwyang** [comment]
yes, you should. change #(label_gt_table_table[i])  to 4 in your case. Also take care of label_gt_tensor_table and label_gt_table_table at line 161 and 162, respectively.

**Comment by zuoeye** [comment]
Is the cvt2TabelLabels() function’s goal to get the numer of target clusters ？Why do we specify the 

numer of target clusters if  the function's goal is to get the numer of target clusters? I still don't 

understand how to specify the number of target clusters?  Could you help me with a detailed 

description of how to modify the code to specify the number of target clusters？

Thank you！

**Comment by zuoeye** [comment]
I just made "target_nclusters_table[i]=4" of line 163. My results were very bad after I run this code in my data. I have changed the 'K_s' from 10 to 100,  'K_c' from 2 to 10 with 1, 'eta' from 0.1 to 0.9, and 'learningRate'. But the results were still bad. Do you know how to improve my results?
